src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -132,13 +132,44 @@ public bool TryCreateContext(string projectUniqueName, out IVsPathContext output
             return outputPathContext != null;
         }
 
-        public bool TryCreateSolutionContext(out IVsPathContext outputPathContext)
+        public bool TryCreateSolutionContext(out IVsPathContext2 outputPathContext)
         {
-            outputPathContext = GetSolutionPathContext();
+            var packagesFolderPath = PackagesFolderPathUtility.GetPackagesFolderPath(_solutionManager.Value, _settings.Value);
+
+            // if solution package folder exists, then set it in VSPathContext
+            if (!string.IsNullOrEmpty(packagesFolderPath) && Directory.Exists(packagesFolderPath))
+            {
+                outputPathContext = new VsPathContext(NuGetPathContext.Create(_settings.Value), packagesFolderPath);
+            }
+            else
+            {
+                outputPathContext = new VsPathContext(NuGetPathContext.Create(_settings.Value));
+            }
 
             return outputPathContext != null;
         }
 
+        public bool TryCreateSolutionContext(string solutionDirectory, out IVsPathContext2 outputPathContext)
+        {
+            if (solutionDirectory == null)
+            {
+                throw new ArgumentNullException(nameof(solutionDirectory));
+            }
+
+            var packagesFolderPath = PackagesFolderPathUtility.GetPackagesFolderPath(solutionDirectory, _settings.Value);
+
+            // if solution package folder exists, then set it in VSPathContext
```

Comment:
The code from here to the end of the function is duplicated in both `TryCreateSolutionContext`, consider merging them in a private method and making both functions call it.